src/amr/level_initializer/hybrid_level_initializer.hpp

```diff
@@ -52,24 +52,24 @@
             auto& hybridModel = static_cast<HybridModel&>(model);
             auto& level       = amr_types::getLevel(*hierarchy, levelNumber);
 
-            auto& hybMessenger = dynamic_cast<HybridMessenger&>(messenger);
+            auto& hybMessenger  = dynamic_cast<HybridMessenger&>(messenger);
+            bool isRegriddingL0 = isRegridding and levelNumber == 0;
 
-            if (isRootLevel(levelNumber))
+            if (isRegridding)
             {
-                PHARE_LOG_START("hybridLevelInitializer::initialize : root level init");
-                model.initialize(level);
-                messenger.fillRootGhosts(model, level, initDataTime);
-                PHARE_LOG_STOP("hybridLevelInitializer::initialize : root level init");
+                std::cout << "regriding level " << levelNumber << "\n";
+                PHARE_LOG_START("hybridLevelInitializer::initialize : regriding block");
+                messenger.regrid(hierarchy, levelNumber, oldLevel, model, initDataTime);
+                PHARE_LOG_STOP("hybridLevelInitializer::initialize : regriding block");
             }
-
             else
             {
-                if (isRegridding)
+                if (isRootLevel(levelNumber))
                 {
-                    std::cout << "regriding level " << levelNumber << "\n";
-                    PHARE_LOG_START("hybridLevelInitializer::initialize : regriding block");
-                    messenger.regrid(hierarchy, levelNumber, oldLevel, model, initDataTime);
-                    PHARE_LOG_STOP("hybridLevelInitializer::initialize : regriding block");
+                    PHARE_LOG_START("hybridLevelInitializer::initialize : root level init");
+                    model.initialize(level);
+                    messenger.fillRootGhosts(model, level, initDataTime);
+                    PHARE_LOG_STOP("hybridLevelInitializer::initialize : root level init");
```

Comment:
how do you know it's not already a configurable logging framework?

Reply:
Our project optionally allows the usage of [LLNL Caliper](https://github.com/LLNL/Caliper)
and if it's active this macro forward to caliper, so it's just our wrapper, to allow with and without caliper.

it could also allow other logging options or frameworks at compile time, so it's quite flexible already